src/dags/common/path.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import shutil
+from contextlib import suppress
+from pathlib import Path
+
+from airflow.decorators import task
+
+logger = logging.getLogger(__name__)
+
+
+@task  # type: ignore[misc]
+def mk_dir(path: Path, clean_if_exists: bool = False) -> Path:
+    """Create and optionally clean directory.
+
+    It is okay for the directory to be created to already exist.
+    The parameter ``clean_if_exists`` can be used
+    to specify that the directory should be cleaned
+    of any containing directories and files.
+    If this cleaning attempt results in errors,
+    they will simply be ignored;
+    The cleaning is simply a best effort attempt.
+
+    Args:
+        path: Path of the directory to create
+        clean_if_exists: Whether to clean the directory if it already existed.
+
+    Returns:
+        The path of the directory created (or that already existed).
+    """
+    logger.debug("Creating directory '%s'.", path)
+    path.mkdir(parents=True, exist_ok=True)
+    if clean_if_exists:
+        # This is a best effort attempt. Hence we are ignoring errors wherever we can.
+        for p in path.iterdir():
+            if p.is_dir():
+                logger.debug("Removing directory '%s'.", p)
```

Comment:
```suggestion
                logger.debug("Removing directory %r", p)
```

Reply:
I choose `%s` specifically as I'd rather see _Removing directory '/tmp/anpr/fubar'_ than _Removing directory 'PosixPath("/tmp/anpr/fubar")'_. Do you prefer the latter? If so why?

Reply:
Oh, right, PosixPath... thing is, when the code that constructs paths is messed up, you can end up with paths containing commas and spaces. I want to be able to spot those.

Reply:
`%r` of `str(p)` prints what I want.

Reply:
`str(p)` in exactly the same as `%s`

* `%s` -> `__str__`
* `%r` -> `__repr__

or am I completely missing what you are saying. Still a little high from my run. ;-)

Reply:
```python
p = Path("foo, bar")

"%s" % p       # "foo, bar"
"%r" % p       # "PosixPath('foo, bar')"
"%r" % str(p)  # "'foo, bar'"
```

Reply:
`%r` of `p.name` might be more readable.

Reply:
Now I get it. Sorry for being a little slow. Changing things to use `%r`.

Reply:
Fixed as part of the original commit. I love my rebases.